### PR TITLE
docs: Contributor Documentation

### PR DESCRIPTION
**Pull request by dmurray-lacework**
issue: https://lacework.atlassian.net/browse/ALLY-660

Improve documentation for contributors by adding CONTRIBUTING.md and issue templates.
Improve experience of using make cmds by adding `make help` and noting *CI ONLY* commands

usage: 
```
❯ make help
-------------------------------------------------------------------
Lacework go-sdk Makefile helper:

 prepare - Initialize the go environment
 test - Run all go-sdk tests
 integration - Build and run integration tests
 integration-only - Run integration tests
 integration-lql - Build and run lql integration tests
 integration-lql-only - Run lql integration tests
 integration-policy - Build and run lql policy tests
 integration-policy-only - Run lql policy tests
 coverage - Output coverage profile information for each function
 coverage-html - Generate HTML representation of coverage profile
 coverage-ci - Generate HTML coverage output for ci pipeline.
 go-vendor - Runs go mod tidy, vendor and verify to cleanup, copy and verify dependencies
 lint - Runs go linter
 fmt - Runs and applies go formatting changes
 fmt-check - Lists formatting issues
 imports-check - Lists imports issues
 build-cli-cross-platform - Compiles the go-sdk for all supported platforms
 generate-databox - *CI ONLY* Generates in memory representation of template files
 generate-docs - *CI ONLY* Generates documentation
 test-resources - *CI ONLY* Prepares CI test containers
 install-cli - Build the go-sdk project and install the resulting binary
 release - *CI ONLY* Prepares a new release of the go-sdk
 install-tools - Install go indirect dependencies
 git-env - Configure git commit message style enforcement by applying git_env.sh
 vagrant-macos-up - Start and provision the vagrant environment: MacOs Sierra
 vagrant-macos-login - Connect to vagrant environment: MacOs Sierra
 vagrant-macos-destroy - Stop and delete vagrant environment: MacOs Sierra
 vagrant-linux-up - Start and provision the vagrant environment: Ubuntu 1804
 vagrant-linux-login - Connect to vagrant environment: Ubuntu 1804
 vagrant-linux-destroy - Stop and delete vagrant environment: Ubuntu 1804
 vagrant-windows-up - Start and provision the vagrant environment: Windows 10
 vagrant-windows-login - Connect to vagrant environment: Windows 10
 vagrant-windows-destroy - Stop and delete vagrant environment: Windows 10
-------------------------------------------------------------------
```

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/pull-request-template.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+name: Pull request template
+about: 'type(scope): Subject of the pull request '
```
Comment (dmurray-lacework) [inline .github/ISSUE_TEMPLATE/pull-request-template.md:3]: title field not about
```suggestion
title: 'type(scope): Subject of the pull request '
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+## Reporting Bugs
+
+Ensure the issue you are raising has not already been created under [issues](https://github.com/lacework/go-sdk/issues).
+
+If no current issue addresses the problem, open a new [issue](https://github.com/lacework/go-sdk/issues/new).
+Include as much relevant information as possible. See the [bug template](https://github.com/lacework/go-sdk/) for help on creating a new issue.
+
+## Feature Requests
+
+If you wish to submit a request to add new functionality or an improvement to the go-sdk then use the the [feature request]() template to
```
Comment (dmurray-lacework) [inline CONTRIBUTING.md:30]: Insert correct link to feature request page
```suggestion
If you wish to submit a request to add new functionality or an improvement to the go-sdk then use the the [feature request](https://github.com/lacework/go-sdk/.github/ISSUE_TEMPLATE/feature_request.md) template to 
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+Include as much relevant information as possible. See the [bug template](https://github.com/lacework/go-sdk/) for help on creating a new issue.
+
+## Feature Requests
+
+If you wish to submit a request to add new functionality or an improvement to the go-sdk then use the the [feature request]() template to 
+open a new [issue](https://github.com/lacework/go-sdk/issues/new)
+
+## Creating a Pull Request
+
+When submitting a pull request follow the [commit message standard](#commit-message-standard).
```
Comment (dmurray-lacework) [inline CONTRIBUTING.md:35]: Feel this needs fleshed out more, instead of jumping straight to commit messages

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+Read the [README.md](https://github.com/lacework/go-sdk/wiki/CLI-Documentation), the [development documentation](https://github.com/lacework/go-sdk/tree/main/cli#development) 
+and the Lacework CLI documentation on the [wiki](https://github.com/lacework/go-sdk/wiki/CLI-Documentation).
+
+## Reporting Bugs
+
+Ensure the issue you are raising has not already been created under [issues](https://github.com/lacework/go-sdk/issues).
+
+If no current issue addresses the problem, open a new [issue](https://github.com/lacework/go-sdk/issues/new).
+Include as much relevant information as possible. See the [bug template](https://github.com/lacework/go-sdk/) for help on creating a new issue.
```
Comment (dmurray-lacework) [inline CONTRIBUTING.md:26]: correct link to bug template

```suggestion
Include as much relevant information as possible. See the [bug template](https://github.com/lacework/go-sdk/.github/ISSUE_TEMPLATE/bug_report.md) for help on creating a new issue.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+* [Feature Requests](#feature-requests)
+* [Pull Requests](#pull-requests)
+
+[Styleguides](#styleguides)
+* [Git Commit Messages](#commit-message-standard)
+
+
+## Before getting started
+
+Read the [README.md](https://github.com/lacework/go-sdk/wiki/CLI-Documentation), the [development documentation](https://github.com/lacework/go-sdk/tree/main/cli#development)
```
Comment (dmurray-lacework) [inline CONTRIBUTING.md:18]: Correct Readme link
```suggestion
Read the [README.md](https://github.com/lacework/go-sdk/blob/main/README.md), the [development documentation](https://github.com/lacework/go-sdk/tree/main/cli#development) 
```

File: Makefile [hunk 6]
```diff
@@ -53,73 +65,92 @@ integration-only:
...
 	@test -z $(shell gofmt -l $(shell go list -f {{.Dir}} ./...))
 
-imports-check:
+.PHONY: imports-check
+imports-check: ## Lists imports issues
 	@test -z $(shell goimports -l $(shell go list -f {{.Dir}} ./...))
 
-build-cli-cross-platform:
+.PHONY: build-cli-cross-platform
+build-cli-cross-platform: ## Compiles the go-sdk for all supported platforms
```
Comment (afiune) [inline Makefile:119]:
```suggestion
build-cli-cross-platform: ## Compiles the Lacework CLI for all supported platforms
```

File: Makefile [hunk 7]
```diff
@@ -53,73 +65,92 @@ integration-only:
...
 	go generate cli/cmd/docs.go
 
-test-resources:
+.PHONY: test-resources
+test-resources: ## *CI ONLY* Prepares CI test containers
 	scripts/prepare_test_resources.sh all
 
-install-cli: build-cli-cross-platform
+.PHONY: install-cli
+install-cli: build-cli-cross-platform ## Build the go-sdk project and install the resulting binary
```
Comment (afiune) [inline Makefile:140]:
```suggestion
install-cli: build-cli-cross-platform ## Build and install the Lacework CLI binary at /usr/local/bin/lacework
```

File: api/README.md [hunk 8]
```diff
@@ -58,3 +58,7 @@ func main() {
 ```
 
 Look at the [examples/](examples/) folder for more examples.
+
+## Contributing
+
+For guidlines on how to contribute to the project see the [CONTRIBUTING.md](CONTRIBUTING.md)
```
Comment (afiune) [inline api/README.md:64]:
```suggestion
For guidelines on how to contribute to the project see the [CONTRIBUTING.md](CONTRIBUTING.md)
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+When submitting a pull request follow the [commit message standard](#commit-message-standard).
+Reduce the likelihood of pushing breaking changes by running the go-sdk unit and integration tests, 
+see [development documentation](https://github.com/lacework/go-sdk/tree/main/cli#development).
+
+## Commit message standard
+
+The format is:
+type(scope): subject
+BODY
+FOOTER
```
Comment (afiune) [inline CONTRIBUTING.md:44]: I think you might need some carrier returns, please check that is well rendered
```suggestion

type(scope): subject

BODY

FOOTER
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+name: Feature request
+about: Suggest an idea for this project
+title: 'feat: '
+labels: 'feat'
+---
+
+# Feature Request
+
+**Describe the Feature Request**
+<!-- A clear and concise description of what the feature request is. Please include if your feature request is related to a problem -->
```
Comment (afiune) [inline .github/ISSUE_TEMPLATE/feature_request.md:11]: Why these comments are surrounded by `<!-- -->`

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+[Before getting started?](#before-getting-started)
+
+[How to Contribute?](#how-can-i-contribute)
+* [Reporting Bugs](#reporting-bugs)
+* [Feature Requests](#feature-requests)
+* [Pull Requests](#pull-requests)
+
+[Styleguides](#styleguides)
+* [Git Commit Messages](#commit-message-standard)
+
```
Comment (afiune) [inline CONTRIBUTING.md:14]: this looks better
```suggestion
### Table Of Contents

1. [Before getting started?](#before-getting-started)
1. [How to Contribute?](#how-can-i-contribute)
    * [Reporting Bugs](#reporting-bugs)
    * [Feature Requests](#feature-requests)
    * [Pull Requests](#pull-requests)
1. [Styleguides](#styleguides)
    * [Git Commit Messages](#commit-message-standard)
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+#### Table Of Contents
+
+[Before getting started?](#before-getting-started)
+
+[How to Contribute?](#how-can-i-contribute)
+* [Reporting Bugs](#reporting-bugs)
+* [Feature Requests](#feature-requests)
+* [Pull Requests](#pull-requests)
+
+[Styleguides](#styleguides)
```
Comment (afiune) [inline CONTRIBUTING.md:12]: There is no section for this, will you add it?

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
+# Contributing to the Lacework Go-sdk
+
+#### Table Of Contents
+
+[Before getting started?](#before-getting-started)
+
+[How to Contribute?](#how-can-i-contribute)
+* [Reporting Bugs](#reporting-bugs)
+* [Feature Requests](#feature-requests)
+* [Pull Requests](#pull-requests)
```
Comment (afiune) [inline CONTRIBUTING.md:10]: There is no section for this, will you add it?

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
+# Contributing to the Lacework Go-sdk
+
+#### Table Of Contents
+
+[Before getting started?](#before-getting-started)
+
+[How to Contribute?](#how-can-i-contribute)
```
Comment (afiune) [inline CONTRIBUTING.md:7]: There is no section for this, will you add it?

File: CONTRIBUTING.md (hunk 11 above)
Comment (afiune) [reply CONTRIBUTING.md:14]: another suggestion would be to use bullets
```suggestion
### Table Of Contents

- [Before getting started?](#before-getting-started)
- [How to Contribute?](#how-can-i-contribute)
	* [Reporting Bugs](#reporting-bugs)
  * [Feature Requests](#feature-requests)
  * [Pull Requests](#pull-requests)
- [Styleguides](#styleguides)
	* [Git Commit Messages](#commit-message-standard)
```

**Review by afiune — APPROVED** [review]

**Pull request merged**